### DISCUSSION TITLE
Is there a way to get Webhooks for on editing completion?

### DISCUSSION THREAD

**Discussion by jdperos** [outline/outline#5726] [General] [answered]
Hello,

I am attempting to use Webhooks (with the help of n8n) to notify our development discord server when Outline documents are updated. Unfortunately, due to the autosave feature, document.updated is spammed every few seconds that someone is modifying a wiki page. I'm hoping to only have a trigger (or filter) to only send this notification when the user clicks "Done Editing" (we're not using Seamless Editing)

What is my best option here?

**Comment by tommoor** [comment]
Use the webhook for `"revisions.create"` instead of `"documents.update"` – it should be what you're after.

**Reply by jdperos** [reply]
@tommoor thanks for the reply. Unfortunately, it seems that `revisions.create` is missing a lot of the information that `documents.update` has that I'm looking for - things like the model's url, and the person who had updated (revised in this case) the page. It's also firing off every time a document is published, too, which I think I can filter around, but isn't ideal.

**Reply by tommoor** [reply]
Not ideal, the best option for now is to consume that event and then query the API based on `revision.documentId` to get the other details

**Reply by tommoor** [reply, marked as answer]
Actually, better answer:

- The person that performed the update is available under `revision.createdBy`
- If you need the "real" document url you do still need to query the API but you can also construct a url in the form `<hostname>/doc/<documentId>` and that will correctly redirect to the right document.

**Reply by jdperos** [reply]
@tommoor  That did it. The revision.create json didn't have the createdBy itself, but it did have a revision ID, which I could query with /revisions.info. So between that HTTP request and the /document.info HTTP request I was able to piece together what I needed. Sure could be simplified with a little bit of extra info in the revision.create webhook though. :)

**Reply by tommoor** [reply]
I'll take a look at that, the user should be there so it feels like a bug

**Reply by tommoor** [reply]
Just checked and the field is there in the payload for me…

<img width="547" alt="image" src="https://github.com/outline/outline/assets/380914/1854f6e0-f630-4b3a-9f3c-e885034730ef">

**Reply by jdperos** [reply]
@tommoor I was assuming that the `createdBy` property was referring to the model, so the document itself. Not the creator of the revision?

**Reply by tommoor** [reply]
A revision is considerd to be it's own model, so in this case it's the creator of that revision not the document

**Reply by jdperos** [reply]
Oh I see. That helps. So in that case I can save the /revisions.info call to get the user. Still need to do the /document.info call to get the URL from the `documentId`. Am I understanding this correctly?

**Reply by tommoor** [reply]
Yes, but you can construct a url that will work from the `revision.documentId` as mentioned in my first answer and avoid that call too.

**Reply by jdperos** [reply]
Awesome. Thanks for the help!
